domain_entity.module

```diff
@@ -491,3 +493,33 @@ function _domain_entity_get_related_domains(EntityInterface $entity) {
 
   return $domains;
 }
+
+/**
+ * Implements hook_domain_references_alter().
+ */
+function domain_entity_domain_references_alter($query, $account, $context) {
+
+  if ($account->hasPermission('publish to any domain')) {
+    return;
+  }
+
+  $allowed = \Drupal::service('domain_access.manager')->getAccessValues($account);
+  $query->condition('id', array_keys($allowed), 'IN');
```

Comment:
Please add code comment a kind of `// Limit list of domains to a set that user can use.`